### ISSUE TITLE
Cannot set timezone - Read-only file system

### ISSUE THREAD

**Issue by TimDowker** [roleoroleo/yi-hack-Allwinner#8]
Trying to set the appropriate timezone and receive this error after trying to run the command 

`echo TimeZone > /etc/TZ`

-ash: can't create /etc/TZ: Read-only file system

Found at 
[https://github.com/TheCrypt0/yi-hack-v4/wiki/Setting-the-Timezone](https://github.com/TheCrypt0/yi-hack-v4/wiki/Setting-the-Timezone)

**Comment by roleoroleo** [comment]
The rootfs is read only.
It's different from the previous cams.
To solve this issue we should rewrite the rootfs increasing the risk of brick.

**Comment by gmarfjan** [comment]
I've done an ugly fix in /home/yi-hack/script/system.sh by adding `export TZ="GMT-2"` before starting all services (GMT-2 is my timezone). So all process start with this environment variable. Now watermark on snapshot displays the right date. When I'm connected by ssh, I have a wrong date because TZ is not set (should be set in /etc/profile but the fs is read-only). When I set it manually, it looks good.
We could add TZ parameter in /home/yi-hack/etc/system.conf and add `export TZ=$(get_config TZ)` in  /home/yi-hack/script/system.sh . Web UI should be modified by adding a droplist of all timezones.
Do you think that's a good way ?

**Comment by roleoroleo** [comment]
In this maner, only "hacked" services start with new TZ.

Yi services start without it.

If it is enough it could be a good solution

**Comment by gmarfjan** [comment]
That's not the right way. For completely resolve this issue, I agree with you, we have to fix the TZ on OS level. It's just a temporary fix.

**Comment by TimDowker** [comment]
Thank you for adding the setting in the config.  However it doesn't seem to be retaining the setting.

**Comment by roleoroleo** [comment]
I retested it now and it seems to work.

**Comment by TimDowker** [comment]
What timezone are you using?  I'm trying America/Toronto which is 

"EST5EDT,M3.2.0,M11.1.0"

And it doesn't work.  Also tried just "EST5EDT" and same thing.

**Comment by roleoroleo** [comment]
CET-1CEST,M3.5.0,M10.5.0/3

**Comment by TimDowker** [comment]
Still no go.  What can I do to help diagnose this?

**Comment by roleoroleo** [comment]
Check if the file /home/yi-hack/etc/system.conf contains the item "TIMEZONE".

**Comment by TimDowker** [comment]
It does.

`root@yi-b73d:/home/yi-hack/etc# cat system.conf 
HTTPD=yes
TELNETD=no
SSHD=yes
FTPD=no
BUSYBOX_FTPD=no
DISABLE_CLOUD=yes
REC_WITHOUT_CLOUD=yes
MQTT=yes
RTSP=yes
RTSP_STREAM=both
ONVIF=no
ONVIF_PROFILE=both
NTPD=yes
NTP_SERVER=pool.ntp.org
PROXYCHAINSNG=no
RTSP_PORT=554
RTSP1_PORT=8554
ONVIF_PORT=80
HTTPD_PORT=8080
USERNAME=
PASSWORD=
FREE_SPACE=10
ONVIF_WM_SNAPSHOT=no
TIMEZONE=
root@yi-b73d:/home/yi-hack/etc#`

**Comment by roleoroleo** [comment]
Try to add a new line after TIMEZONE=

**Comment by TimDowker** [comment]
Ok, so adding the line didn't work but I was able to manually add it to system.conf.  It still doesn't show anything in the "Timezone" text box on the Configurations page however the correct time is shown on the main information page.

**Comment by TimDowker** [comment]
`date` doesn't return the correct date though.

**Comment by roleoroleo** [comment]
I can't understand...

**Comment by roleoroleo** [comment]
I restored 0.1.3 and upgraded again to 0.1.4.
I have no problems.

**Comment by roleoroleo** [comment]
Has anyone tested this functionality with version 0.1.4?

**Comment by emeins** [comment]
I tested with version 1.4.0. I added the TZ string (CET-1CEST,M3.5.0,M10.5.0/3) in the general settings of the web interface and it works well. I can see the TIMEZONE=CET-1CEST,M3.5.0,M10.5.0/3 in the /home/yi-hack/etc/system.conf without adding additional lines manually. I was a bit confused when I checked with #date command in ssh shell - still GMT time. But the snapshot and motion events present the right timezone.
